tests/test_lag_2.py

```diff
@@ -40,244 +35,239 @@ def common_setup_teardown(duthost, ptfhost, testbed, conn_graph_facts):
+            lag_rate_current_setting = None
+
+            # Get the vm host(veos) by it host name
+            vm_host = self.nbrhosts[peer_device]
+
+            # Make sure all lag members on VM are set to fast
+            logging.info("Changing lacp rate to fast for %s" % neighbor_lag_intfs[0])
+            vm_host.set_interface_lacp_rate_mode(neighbor_lag_intfs[0], 'fast')
+            lag_rate_current_setting = 'fast'
+            time.sleep(5)
+            for iface_behind_lag in iface_behind_lag_member:
+                self.__verify_lag_lacp_timing(1, iface_behind_lag)
+
+            # Make sure all lag members on VM are set to slow
+            vm_host.set_interface_lacp_rate_mode(neighbor_lag_intfs[0], 'normal')
+            lag_rate_current_setting = 'slow'
+            time.sleep(5)
+            for iface_behind_lag in iface_behind_lag_member:
+                self.__verify_lag_lacp_timing(30, iface_behind_lag)
+        finally:
+            # Restore lag rate setting on VM in case of failure
+            if lag_rate_current_setting == 'fast':
+                vm_host.set_interface_lacp_rate_mode(neighbor_lag_intfs[0], 'normal')
+
+    def run_single_lag_test(self, lag_name):
+        logging.info("Start checking single lag for: %s" % lag_name)
+
+        lag_facts           = self.__get_lag_facts()
+        intf, po_interfaces = self.__get_lag_intf_info(lag_facts, lag_name)
+        po_flap             = self.__check_flap(lag_facts, lag_name)
+
+        # Figure out fanout switches info if exists for the lag member and run minlink test
+        if intf in self.fanout_neighbors.keys():
+            peer_device   = self.fanout_neighbors[intf]['peerdevice']
+            neighbor_intf = self.fanout_neighbors[intf]['peerport']
+            self.__verify_lag_minlink(self.fanouthosts[peer_device], lag_name, intf, neighbor_intf, po_interfaces, po_flap, deselect_time=5)
+
+        # Figure out remote VM and interface info for the lag member and run minlink test
+        peer_device   = self.vm_neighbors[intf]['name']
+        neighbor_intf = self.vm_neighbors[intf]['port']
+        self.__verify_lag_minlink(self.nbrhosts[peer_device], lag_name, intf, neighbor_intf, po_interfaces, po_flap, deselect_time=95)
+
+    def run_lag_fallback_test(self, lag_name):
+        logging.info("Start checking lag fall back for: %s" % lag_name)
+
+        lag_facts           = self.__get_lag_facts()
+        intf, po_interfaces = self.__get_lag_intf_info(lag_facts, lag_name)
+        po_fallback         = lag_facts['lags'][lag_name]['po_config']['runner']['fallback']
+
+        # Figure out remote VM and interface info for the lag member and run lag fallback test
+        peer_device   = self.vm_neighbors[intf]['name']
+        neighbor_intf = self.vm_neighbors[intf]['port']
+        vm_host       = self.nbrhosts[peer_device]
+
+        try:
+            # Shut down neighbor interface
+            vm_host.shutdown(neighbor_intf)
+            time.sleep(120)
```

Comment:
Can we use wailt_until here instead of 2 minutes sleep?

Reply:
Yes. My previous pytest converting(config_test, seems like it has been renamed in another folder) also used the magic wait, I will address the code change for that test case too.